### DISCUSSION TITLE
last 4 bits of incoming pdu is truncated on purpose?

### DISCUSSION THREAD

**Discussion by Wahib13** [python-smpplib/python-smpplib#150] [Q&A] [answered]
so I was testing receiving deliver_sm as a client and noticed a custom parameter I had introduced had the last 4 characters missing always. and I came across this line in the client.py code:
     raw_pdu = self._socket.recv(length - 4)

I have updated it to 
    raw_pdu = self._socket.recv(length)
and now the full parameter data is received alright. but does anyone know why it was there in the first place? I don't want to launch and have problems later. 
or if it is okay like this then I will leave it like that.

**Comment by eigenein** [comment, marked as answer]
Hi @Wahib13,

`length` includes the length field itself, which is 4 bytes. So that the length field is already read, we have to read the rest `length - 4` bytes.

**Comment by Wahib13** [comment]
okay will run some more tests on Monday to confirm. thanks for the reply
